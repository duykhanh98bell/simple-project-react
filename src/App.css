.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@import url('https://fonts.googleapis.com/css?family=Poppins:400,500,600,700&display=swap');
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Poppins', sans-serif;
}
.wrapper {
  height: 100%;
  /* width: 300px; */
  position: relative;
}

.wrapper .menu-btn i {
  /* position: absolute; */
  font-size: 23px;
  transition: all 0.3s ease;
}
/* .wrapper .menu-btn i.fa-times {
  opacity: 0;
} */
/* #btn:checked ~ .menu-btn i.fa-times {
  opacity: 1;
  transform: rotate(-180deg);
}
#btn:checked ~ .menu-btn i.fa-bars {
  opacity: 0;
  transform: rotate(180deg);
} */
#sidebar {
  /* position: fixed; */
  background: #404040;
  height: 100%;
  /* width: 100%; */
  /* overflow: hidden; */
  /* left: -270px; */
  /* transition: all 0.3s ease; */
}
#btn:checked ~ #sidebar {
  left: 0;
}
#sidebar .title {
  line-height: 65px;
  text-align: center;
  background: #333;
  font-size: 25px;
  font-weight: 600;
  color: #f2f2f2;
  border-bottom: 1px solid #222;
}
#sidebar .list-items {
  position: relative;
  background: #404040;
  width: 100%;
  height: 100%;
  list-style: none;
}
#sidebar .list-items li {
  padding-left: 40px;
  line-height: 50px;
  border-top: 1px solid rgba(255, 255, 255, 0.1);
  border-bottom: 1px solid #333;
  transition: all 0.3s ease;
}
#sidebar .list-items li:hover {
  border-top: 1px solid transparent;
  border-bottom: 1px solid transparent;
  box-shadow: 0 0px 10px 3px #222;
}
#sidebar .list-items li:first-child {
  border-top: none;
}
#sidebar .list-items li div {
  text-align: start;
  color: #f2f2f2;
  text-decoration: none;
  font-size: 18px;
  font-weight: 500;
  height: 100%;
  width: 100%;
  display: block;
}
#sidebar .list-items li div i {
  margin-right: 20px;
}
/* #sidebar .list-items .icons {
  width: 100%;
  height: 40px;
  text-align: center;
  position: absolute;
  bottom: 100px;
  line-height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
} */
/* #sidebar .list-items .icons button {
  height: 100%;
  width: 40px;
  display: block;
  margin: 0 5px;
  font-size: 18px;
  color: #f2f2f2;
  background: #4a4a4a;
  border-radius: 5px;
  border: 1px solid #383838;
  transition: all 0.3s ease;
} */
/* #sidebar .list-items .icons button:hover {
  background: #404040;
}
.list-items .icons button:first-child {
  margin-left: 0px;
} */
/* .content {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #202020;
  z-index: -1;
  width: 100%;
  text-align: center;
}
.content .header {
  font-size: 45px;
  font-weight: 700;
}
.content p {
  font-size: 40px;
  font-weight: 700;
} */
