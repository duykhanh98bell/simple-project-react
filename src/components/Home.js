import React, { useState, useEffect } from 'react';
import Employee from './employee/Employee';
import Department from './department/Department';
import Register from './auth/Register';
import axios from 'axios';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';

export const Home = (props) => {
  const [department, setDepartment] = useState(null);
  const [oneDepart, setOneDepart] = useState('');
  const [employee, setEmployee] = useState(null);
  const [oneEmployee, setOneEmployee] = useState('');
  const [employeeIn, setEmployeeIn] = useState(null);

  useEffect(() => {
    getDepartment().then((res) => setDepartment(res));
    getEmployee().then((res) => setEmployee(res));
  }, []); // gọi list trước khi load trang để có dữ liệu hiển thị

  //  Start Department
  const getDepartment = () =>
    axios
      .get('/department', {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => res.data)
      .catch((err) => err);

  const addDepartment = (payload) => {
    axios
      .post('/department/', payload, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => console.log(res.data))
      .catch((err) => err)
      .then(() => getDepartment().then((res) => setDepartment(res)));
  };

  const onDelete = (id) => {
    axios
      .delete('/department/' + id, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => console.log(res.data))
      .catch((err) => err)
      .then(() => getDepartment().then((res) => setDepartment(res)));
  };

  const findDepart = (id) => {
    axios
      .get('/department/' + id, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => setOneDepart(res.data))
      .catch((err) => err);
  };

  const saveUpdate = (payload) => {
    axios
      .put(
        '/department/' + payload._id,
        {
          name: payload.name,
          officephone: payload.officephone,
          manager: payload.manager,
        },
        {
          headers: {
            Authorization: 'Bearer ' + localStorage.getItem('token'),
          },
        }
      )
      .then((res) => console.log(res.data.message))
      .catch((err) => err)
      .then(() => getDepartment().then((res) => setDepartment(res)));
  };
  // End Department

  // Start Employee
  const getEmployee = () =>
    axios
      .get('/employee', {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => res.data.all)
      .catch((err) => err);

  const addEmployee = async (data, filename) => {
    const formData = new FormData();
    formData.append('photo', filename);
    formData.append('name', data.name);
    formData.append('jobtitle', data.jobtitle);
    formData.append('cellphone', data.cellphone);
    formData.append('email', data.email);
    formData.append('department_id', data.department_id);

    const config = {
      headers: {
        'content-type': 'multipart/form-data',
        Authorization: 'Bearer ' + localStorage.getItem('token'),
      },
    };
    try {
      const res = await axios
        .post('/employee/', formData, config)
        .then((res) => console.log(res.data))
        .catch((err) => err)
        .then(() => getEmployee().then((res) => setEmployee(res)));
      return res;
    } catch (err) {
      return err.response;
    }
  };

  const DeleteEmp = async (id) => {
    try {
      const res = await axios
        .delete('/employee/' + id, {
          headers: {
            Authorization: 'Bearer ' + localStorage.getItem('token'),
          },
        })
        .then((res) => console.log(res.data))
        .catch((err) => err)
        .then(() => getEmployee().then((res) => setEmployee(res)));
      return res.data.message;
    } catch (err) {
      return err;
    }
  };

  const idEm = (id) => {
    axios
      .get('/employee/' + id, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => setOneEmployee(res.data))
      .catch((err) => err);
  };

  const saveEditEmployee = async (payload, filename) => {
    const formData = new FormData();
    if (filename !== null) {
      formData.append('photo', filename);
    }
    formData.append('name', payload.name);
    formData.append('cellphone', payload.cellphone);
    formData.append('department_id', payload.department_id);
    formData.append('email', payload.email);
    formData.append('jobtitle', payload.jobtitle);
    const config = {
      headers: {
        'content-type': 'multipart/form-data',
        Authorization: 'Bearer ' + localStorage.getItem('token'),
      },
    };
    try {
      const res = await axios
        .put('/employee/' + payload.id, formData, config)
        .then((res) => console.log(res.data))
        .then(() => getEmployee().then((res) => setEmployee(res)))
        .catch((err) => err);
      return res;
    } catch (err) {
      return err;
    }
  };
  // End Employee
  const findEmployeIn = (id) => {
    axios
      .get('/department/employeein/' + id, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => setEmployeeIn(res.data))
      .catch((err) => err);
  };

  const out = () => {
    axios
      .delete('/auth/logout', {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => {
        localStorage.removeItem('token');
        console.log(res.data);
      })
      .catch((err) => err);
    props.outLogin();
  };
  return (
    <Router>
      <div className='row'>
        <div className='col-2'>
          <div className='wrapper'>
            <nav id='sidebar'>
              <div className='title'>Side Menu</div>
              <ul className='list-items'>
                <li>
                  <div type='button' className='btn'>
                    <i className='fas fa-home'></i>
                    <Link to='/employee'>Employee</Link>
                  </div>
                </li>

                <li>
                  <div type='button' className='btn'>
                    <i className='fas fa-sliders-h'></i>
                    <Link to='/department'>Department </Link>
                  </div>
                </li>

                <li>
                  <div type='button' className='btn'>
                    <i className='fas fa-user'></i>
                    <Link to='/register'>New User </Link>
                  </div>
                </li>
                <li>
                  <div type='button' className='btn' onClick={out}>
                    <i className='fas fa-user'></i>
                    <Link to='/'>Logout </Link>
                  </div>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        <Switch>
          <Route path='/employee'>
            <Employee
              listEmployee={employee}
              listDepartment={department}
              dataEmployee={addEmployee}
              onDeleteEmp={DeleteEmp}
              sendIdEm={idEm}
              sendOneEm={oneEmployee}
              dataEditEmployee={saveEditEmployee}
            />
          </Route>
          <Route path='/department'>
            <Department
              listDepartment={department}
              postDepartment={addDepartment}
              onDelete={onDelete}
              editDepart={findDepart}
              oneDepart={oneDepart}
              postUpdateDepartment={saveUpdate}
              employeeIn={findEmployeIn}
              listEmIn={employeeIn}
            />
          </Route>
          <Route path='/register'>
            <Register />
          </Route>
        </Switch>
      </div>
    </Router>
  );
};

export default Home;
