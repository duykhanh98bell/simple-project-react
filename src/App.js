import React, { useState } from 'react';
import './App.css';
import Home from './components/Home';
import axios from 'axios';
import Login from './components/auth/Login';
import Change from './components/auth/Change';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

function App() {
  const [isLogin, setIsLogin] = useState(0);
  const postLogin = (user) => {
    localStorage.setItem('username', user.username);
    axios
      .post('/auth/login', user)
      .then((res) => {
        if (res.status === 201 && res.data.first === true) {
          localStorage.setItem('token', res.data.access_token);
          setIsLogin(1);
        } else if (res.status === 201 && res.data.first === false) {
          localStorage.setItem('token', res.data.access_token);
          setIsLogin(2);
        }
      })
      .catch((err) => err);
  };
  const postChange = (payload) => {
    axios
      .put('/auth/change', payload, {
        headers: {
          Authorization: 'Bearer ' + localStorage.getItem('token'),
        },
      })
      .then((res) => {
        if (res.status === 200) setIsLogin(0);
      })
      .catch((err) => err);
  };

  const checkPage = () => {
    if (isLogin === 0) {
      return <Login dataLogin={postLogin} />;
    } else if (isLogin === 1) {
      return <Home outLogin={() => setIsLogin(0)} />;
    } else {
      return <Change changePass={postChange} />;
    }
  };

  return (
    <Router>
      <Switch>
        <Route path='/'>{checkPage()}</Route>
      </Switch>
    </Router>
  );
}
export default App;
