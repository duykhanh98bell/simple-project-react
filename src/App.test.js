import { create, act } from 'react-test-renderer';
import { renderHook } from '@testing-library/react-hooks';
import Home from './components/Home';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('home', () => {
  it('render home', () => {
    const { getAllByText, getByText, getByTestId, container } = render(<App />);
    screen.debug();
  });
});
