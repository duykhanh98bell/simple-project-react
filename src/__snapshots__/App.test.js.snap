// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="body"
>
  <div
    className="wrapperr"
  >
    <div
      className="title"
    >
      Login Form
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <input
          name="username"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label>
          Username
        </label>
      </div>
      <div
        className="field"
      >
        <input
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label>
          Password
        </label>
      </div>
      <div
        className="content"
      >
        <div
          className="checkbox"
        >
          <input
            id="remember-me"
            type="checkbox"
          />
          <div
            htmlFor="remember-me"
          >
            Remember me
          </div>
        </div>
        <div
          className="pass-link"
        >
          <div>
            Forgot password?
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <input
          defaultValue="Login"
          type="submit"
        />
      </div>
    </form>
  </div>
</div>
`;
